src/Vis/Visualization.jl

```diff
@@ -560,35 +562,37 @@ end
 Draw the surface representing `raygen`, as well as some sample [`Ray`](@ref)s eminating from it (providing `norays` is false).
 """
 function draw!(scene::MakieLayout.LScene, raygen::GeometricRayGenerator{T}; position::SVector{3,T} = SVector{3,T}(0, 0, 0), norays::Bool = false, kwargs...) where {T<:Real}
-    if !norays
-        for r in raygen
-            draw!(scene, Ray(origin(r) + position, direction(r)); kwargs...)
-        end
-    end
-    draw!(scene, origingen(raygen), position = position, color = :white)
-end
-
-function draw!(scene::MakieLayout.LScene, pixel::PixelSource{T,C}; kwargs...) where {T<:Real,C}
-    for (i, raygen) in enumerate(pixel.subpixels)
-        draw!(scene, raygen; kwargs...)
-    end
-end
-
-function draw!(scene::MakieLayout.LScene, display::BasicDisplayPanel{T,C}; kwargs...) where {T<:Real,C}
-    draw!(scene, display.generator; kwargs...)
+   throw(ErrorException("needs to be rewritten once we figure out what origingen is doing"))
```

Comment:
In general this was because you could position each generator without positioning the sub parts (e.g. move a display to (0,2,3) without having to start by putting the pixel at (0,2,3) - this meant drawing rays became complicated as they had to be transformed for each wrapping class. This also meant splitting out drawing of the actual source panel/circle/point `origingen` just returns the origin generator of the source so that it can drawn directly (without any rays)

This was always a huge mess, but at the time we just wanted to get visualisations working without rewriting everything to do with emitters (now done - hooray!) so hopefully a lot of this code can go

Reply:
deleting it all for now